modules/Emails/EmailsController.php

```diff
@@ -154,6 +161,116 @@ public function action_send()
         $this->view = 'sendemail';
     }
 
+    /**
+     * Parse and replace bean variables
+     * but first validate request,
+     * see log to check validation problems
+     *
+     * return Email bean
+     *
+     * @param Email $email
+     * @param array $request
+     * @return Email
+     */
+    protected function replaceEmailVariables(Email $email, $request) {
+
+        // request validation before replace bean variables
+
+        if ($this->isValidRequestForReplaceEmailVariables($request)) {
+
+            $macro_nv = array();
```

Comment:
In fact I have no idea what does it mean, I just want to use consistent nameing for variables like:
modules/EmailMan.php:329,333,444,773,786,789,790,877
modules/EmailTemplate.php:416,493
test/test/modules/EmailTemplates/EmailTemplateTest.php:127,132